deepmd/dpmodel/descriptor/dpa1.py

```diff
@@ -0,0 +1,911 @@
+    axis_neuron: int
+            Number of the axis neuron :math:`M_2` (number of columns of the sub-matrix of the embedding matrix)
+    tebd_dim: int
+            Dimension of the type embedding
+    tebd_input_mode: str
+            The way to mix the type embeddings. Supported options are `concat`, `dot_residual_s`.
+    resnet_dt: bool
+            Time-step `dt` in the resnet construction:
+            y = x + dt * \phi (Wx + b)
+    trainable: bool
+            If the weights of embedding net are trainable.
+    type_one_side: bool
+            If 'False', type embeddings of both neighbor and central atoms are considered.
+            If 'True', only type embeddings of neighbor atoms are considered.
+            Default is 'False'.
+    attn: int
+            Hidden dimension of the attention vectors
+    attn_layer: int
+            Number of attention layers
+    attn_dotr: bool
+            If dot the angular gate to the attention weights
+    attn_mask: bool
+            If mask the diagonal of attention weights
+    exclude_types : List[List[int]]
+            The excluded pairs of types which have no interaction with each other.
+            For example, `[[0, 1]]` means no interaction between type 0 and type 1.
+    env_protection: float
+            Protection parameter to prevent division by zero errors during environment matrix calculations.
+    set_davg_zero: bool
+            Set the shift of embedding net input to zero.
+    activation_function: str
+            The activation function in the embedding net. Supported options are |ACTIVATION_FN|
+    precision: str
+            The precision of the embedding net parameters. Supported options are |PRECISION|
+    scaling_factor: float
+            The scaling factor of normalization in calculations of attention weights.
+            If `temperature` is None, the scaling of attention weights is (N_dim * scaling_factor)**0.5
+    normalize: bool
+            Whether to normalize the hidden vectors in attention weights calculation.
+    temperature: float
+            If not None, the scaling of attention weights is `temperature` itself.
+    smooth_type_embedding: bool
+            Whether to use smooth process in attention weights calculation.
+    concat_output_tebd: bool
+            Whether to concat type embedding at the output of the descriptor.
+    spin
+            The old implementation of deepspin (deprecated in the descriptor).
+
+    Limitations
+    -----------
+    The currently implementation does not support the following features
+
+    1. type_one_side == True
+    2. exclude_types != []
+    3. spin is not None
+    4. tebd_input_mode != 'concat'
+
+    References
+    ----------
+    .. [1] Duo Zhang, Hangrui Bi, Fu-Zhi Dai, Wanrun Jiang, Linfeng Zhang, and Han Wang. 2022.
+       DPA-1: Pretraining of Attention-based Deep Potential Model for Molecular Simulation.
+       arXiv preprint arXiv:2208.08236.
+    """
+
+    def __init__(
```

Comment:
you should carefully check and explain which parameters are really in use  and which parameters are kept for compatibility.

Reply:
Docs updated, there are now two kinds of **special** parameters:
1. Implemented in this class but only specific options can be consistent with other backends.(e.g. `attn_mask`(only False), `smooth_type_embedding` (only False) of `DescrptDPA1Compat`  in TF to be consistent.) They are marked in docstr like: 
```
    attn_mask: bool
            (Only support False to keep consistent with other backend references.)
            If mask the diagonal of attention weights
```
2. Not implemented in the class but are only kept for compatibility. (e.g. `scaling_factor`, `normalize`, `temperature` of `DescrptDPA1Compat`  in TF and `attn_mask` of `DescrptDPA1` in PT/DP.) They are marked in docstr like:
```
    normalize: bool
            (Only support True to keep consistent with other backend references.)
            (Not used in this version.)
            Whether to normalize the hidden vectors in attention weights calculation.
```

Reply:
I find the attn_mask == True is not implemented at all, this should be explicitly pointed out in the doc.

Reply:
It was pointed out in the `Limitations` at the end of the doc and will also raise `NotImplementedError` at the front lines of `__init__`. I also added more explicit doc in [add docs for not-used param](https://github.com/deepmodeling/deepmd-kit/pull/3696/commits/9662a2c97528448f6065deaddcc50ac87ffa91cc).